src/script/lua_api/l_object.cpp

```diff
@@ -2539,7 +2539,26 @@ int ObjectRef::l_set_lighting(lua_State *L)
 			lighting.volumetric_light_strength = rangelim(lighting.volumetric_light_strength, 0.0f, 1.0f);
 		}
 		lua_pop(L, 1); // volumetric_light
-}
+
+		lua_getfield(L, 2, "light_intensity");
+		if (lua_istable(L, -1)) {
+			lua_getfield(L, 3, "color_offset");
+			if (lua_istable(L, -1)) {
+				lighting.lightIntensity.colorOffset_rgb.X = getfloatfield_default(L, -1, "r", lighting.lightIntensity.colorOffset_rgb.X);
+				lighting.lightIntensity.colorOffset_rgb.Y = getfloatfield_default(L, -1, "g", lighting.lightIntensity.colorOffset_rgb.Y);
+				lighting.lightIntensity.colorOffset_rgb.Z = getfloatfield_default(L, -1, "b", lighting.lightIntensity.colorOffset_rgb.Z);
+			}
+			lua_pop(L, 1); // color_offset
+			lua_getfield(L, 3, "color_ratio_coef");
+			if (lua_istable(L, -1)) {
+				lighting.lightIntensity.colorRatioCoef_rgb.X = getfloatfield_default(L, -1, "r", lighting.lightIntensity.colorRatioCoef_rgb.X);
+				lighting.lightIntensity.colorRatioCoef_rgb.Y = getfloatfield_default(L, -1, "g", lighting.lightIntensity.colorRatioCoef_rgb.Y);
```

Comment:
these names are bad and way too long

why is it called `lightIntensity` and not just `intensity`?
what purpose does the `_rgb` serve here? I can see that it's a vector, but why actually? We have `SColorf`.
not to mention that the code style says to name `variables_like_this` not `variablesLikeThis`

Reply:
In mentioning `SColorf` -> it was discussed here: https://github.com/minetest/minetest/pull/14091#discussion_r1437865790.

I just found `lightIntensity` more understandable than just `intensity`. No problem to rename.

I use `_rgb` because it contains three coefficients. One for each color. Also, no problem to rename.

Reply:
@appgurueu This is related to our previous discussion.

Reply:
Renamed to sky_light.